### PR TITLE
Feature: nlb zone affinity via r53

### PR DESCRIPTION
**Pull request by szuecs**
feature: set nlb zone affinity for clients
log: log default cross zone and zone affinity config

This enables you to set dns_record.client_routing_policy configuration, such that you can decide for NLB via R53 feature that clients running in the same zone will hit also the same zone of your NLB. This is interesting for cross cluster traffic or traffic that stays in the same cluster but passes the NLB. Valid configurations are:
- availability_zone_affinity
- partial_availability_zone_affinity
- any_availability_zone (default)

see also https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html\#load-balancer-attributes and https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html\#zonal-dns-affinity

### CODE REVIEW

**Comment by szuecs** [comment]
I want to test in a dev cluster before merge

File: aws/cf_template.go [hunk 1]
```diff
@@ -307,6 +307,19 @@ func generateTemplate(spec *stackSpec) (string, error) {
 				Value: cloudformation.String(fmt.Sprintf("%t", spec.nlbCrossZone)),
 			},
 		)
+
+		println("HERE: ", spec.nlbZoneAffinity)
```
Comment (AlexanderYastrebov) [inline aws/cf_template.go:311]: HERE:

File: aws/adapter.go [hunk 2]
```diff
@@ -124,6 +125,10 @@ const (
 	DefaultAlbS3LogsPrefix = ""
 
 	DefaultCustomFilter = ""
+
+	// DefaultZoneAffinity specifies dns_record.client_routing_policy, see also https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html#load-balancer-attributes
+	DefaultZoneAffinity = "any_availability_zone"
```
Comment (AlexanderYastrebov) [inline aws/adapter.go:130]: Maybe highlight that it is safe to change as per https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-loadbalancer-loadbalancerattribute.html#aws-properties-elasticloadbalancingv2-loadbalancer-loadbalancerattribute-properties

**Comment by AlexanderYastrebov** [comment]
LGTM modulo minor comments.

**Comment by AlexanderYastrebov** [comment]
:+1:

File: aws/cf_template.go (hunk 1 above)
Comment (szuecs) [reply aws/cf_template.go:311]: oopsi :)

File: aws/adapter.go (hunk 2 above)
Comment (szuecs) [reply aws/adapter.go:130]: done

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
Default:
image

With zone affinity set and cross zone balancing off:
image

**Pull request merged**